vpd-manager/editor_impl.cpp

```diff
@@ -601,9 +601,15 @@ void EditorImpl::updateKeyword(const Binary& kwdData, uint32_t offset,
         // prevent any data/ECC corruption in case BMC reboots while VPD update.
         enableRebootGuard();
 
-        // TODO: Figure out a better way to get max possible VPD size.
+        auto vpdFileSize = std::filesystem::file_size(vpdFilePath);
```

Comment:
This API can throw on exceptions. Try the overloaded version which does not throw and check for error code.

Reply:
if the program fails to get file_size of the given vpd, why we should not throw an exception and why we should continue further?

Reply:
The code should not continue further in either case. That is why I asked to check for error code and you can have error logging on that basis and return.
Also, you can continue with this API that throws exception but then it would be better to catch that here and have some specific logging to know the exact failure. the same way we have at line 625.
Your call, I am ok either way.